src/App/Tracker/Controller/Hooks/Listeners/AbstractListener.php

```diff
@@ -0,0 +1,210 @@
+	 *
+	 * @return  bool    True if the label already exists
+	 *
+	 * @since   1.0
+	 */
+	protected function checkLabel($hookData, Github $github, Logger $logger, $project, $checkLabel)
+	{
+		// The Github ID if we have a pull or issue so that method can handle both
+		$issueNumber = $this->getIssueID($hookData);
+
+		// Get the labels for the pull's issue
+		try
+		{
+			$labels = $github->issues->get($project->gh_user, $project->gh_project, $issueNumber)->labels;
+		}
+		catch (\DomainException $e)
+		{
+			$logger->error(
+				sprintf(
+					'Error retrieving labels for GitHub item %s/%s #%d - %s',
+					$project->gh_user,
+					$project->gh_project,
+					$issueNumber,
+					$e->getMessage()
+				)
+			);
+
+			return false;
+		}
+
+		// Check if the label present that return true
+		if (count($labels) > 0)
+		{
+			foreach ($labels as $label)
+			{
+				if ($label->name == $checkLabel)
+				{
+					return true;
+				}
+			}
+		}
+
+		// Else return false
+		return false;
+	}
+
+	/**
+	 * Remove Labels
+	 *
+	 * @param   object  $hookData      Hook data payload
+	 * @param   Github  $github        Github object
+	 * @param   Logger  $logger        Logger object
+	 * @param   object  $project       Object containing project data
+	 * @param   array   $removeLabels  The labels to remove
+	 *
+	 * @return  void
+	 *
+	 * @since   1.0
+	 */
+	protected function removeLabels($hookData, Github $github, Logger $logger, $project, $removeLabels)
+	{
+		// The Github ID if we have a pull or issue so that method can handle both
+		$issueNumber = $this->getIssueID($hookData);
+
+		// Only try to remove labels if the array isn't empty
+		if (!empty($removeLabels))
+		{
+			// The foreach is needed as we have no array support on the `removeFromIssue` method
+			foreach ($removeLabels as $removeLabel)
+			{
+				try
+				{
+					$github->issues->labels->removeFromIssue(
+						$project->gh_user, $project->gh_project, $issueNumber, $removeLabel
+					);
+
+					// Post the new label on the object
+					$logger->info(
+						sprintf(
+							'Removed %s label to %s/%s #%d',
+							$removeLabel,
+							$project->gh_user,
+							$project->gh_project,
+							$issueNumber
+						)
+					);
+				}
+				catch (\DomainException $e)
+				{
+					$logger->error(
+						sprintf(
+							'Error removing the %s label from GitHub pull request %s/%s #%d - %s',
+							$removeLabel,
+							$project->gh_user,
+							$project->gh_project,
+							$issueNumber,
+							$e->getMessage()
+						)
+					);
+				}
+			}
+		}
+	}
+
+	/**
+	 * Get the correct issue ID if it is a Pull or Issue
+	 *
+	 * @param   object  $hookData  Hook data payload
+	 *
+	 * @return  string  The Issue/Pull ID
+	 *
+	 * @since   1.0
+	 */
+	protected function getIssueId($hookData)
+	{
+		if (isset($hookData->pull_request->number))
+		{
+			return $hookData->pull_request->number;
+		}
+
+		if (isset($hookData->issue->number))
+		{
+			return $hookData->issue->number;
+		}
+	}
```

Comment:
I think we should return something here if we cannot retrieve numbers from `$hookData`. May be `return null;` and then check this `null` in other methods. Then the docblock should be changed also:

`@return  mixed  The Issue number or null` if no issue number found in hook data.